From 5030a23615a63826809b4db1ffd9e358d553fb6c Mon Sep 17 00:00:00 2001
From: Elvis Lee <kwangwoong.lee@lge.com>
Date: Thu, 5 Nov 2020 18:22:02 +0900
Subject: [PATCH] Correct headers to build with eglfs modules

To build with eglfs module and gbm device integration module,
some headers should consider global path, not local.

This covers eglfs_kms and eglfs_kms_egldeivce device integration.
Other device integrations might be future work if needed.

Task-number: QTBUG-85268
Change-Id: I1bad5fbac99aa79d146e90f88b53519b09254e13
Reviewed-by: Qt CI Bot <qt_ci_bot@qt-project.org>
Reviewed-by: Laszlo Agocs <laszlo.agocs@qt.io>
Reviewed-by: Elvis Lee <kwangwoong.lee@lge.com>
Upstream-Status: Backport [https://codereview.qt-project.org/c/qt/qtbase/+/320458]
---
 .../eglfs/deviceintegration/eglfs_kms/eglfs_kms_gbm_support.pro | 2 --
 .../eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmdevice.cpp    | 2 +-
 .../eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmdevice_p.h    | 2 +-
 .../eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen.cpp    | 2 +-
 .../eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen_p.h    | 2 +-
 .../eglfs_kms_egldevice/eglfs_kms_egldevice.pro                 | 2 --
 .../deviceintegration/eglfs_kms_egldevice/qeglfskmsegldevice.h  | 2 +-
 .../eglfs_kms_egldevice/qeglfskmsegldeviceintegration.h         | 2 +-
 .../eglfs_kms_egldevice/qeglfskmsegldevicescreen.h              | 2 +-
 .../deviceintegration/eglfs_kms_support/eglfs_kms_support.pro   | 2 --
 .../deviceintegration/eglfs_kms_support/qeglfskmsscreen.cpp     | 2 +-
 src/plugins/platforms/eglfs/eglfs-plugin.pro                    | 2 --
 src/plugins/platforms/eglfs/qeglfsmain.cpp                      | 2 +-
 13 files changed, 9 insertions(+), 17 deletions(-)

diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/eglfs_kms_gbm_support.pro b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/eglfs_kms_gbm_support.pro
index a0ff6bce39..38dfcbd386 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/eglfs_kms_gbm_support.pro
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/eglfs_kms_gbm_support.pro
@@ -4,8 +4,6 @@ load(qt_module)
 
 QT += core-private gui-private eglfsdeviceintegration-private eglfs_kms_support-private kms_support-private edid_support-private
 
-INCLUDEPATH += $$PWD/../../api $$PWD/../eglfs_kms_support
-
 # Avoid X11 header collision, use generic EGL native types
 DEFINES += QT_EGL_NO_X11
 
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmdevice.cpp b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmdevice.cpp
index 8c653cb4c2..9a627db49d 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmdevice.cpp
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmdevice.cpp
@@ -42,7 +42,7 @@
 #include "qeglfskmsgbmdevice_p.h"
 #include "qeglfskmsgbmscreen_p.h"
 
-#include "qeglfsintegration_p.h"
+#include <private/qeglfsintegration_p.h>
 
 #include <QtCore/QLoggingCategory>
 #include <QtCore/private/qcore_unix_p.h>
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmdevice_p.h b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmdevice_p.h
index 701727a7e6..8e91b806c5 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmdevice_p.h
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmdevice_p.h
@@ -54,7 +54,7 @@
 //
 
 #include "qeglfskmsgbmcursor_p.h"
-#include <qeglfskmsdevice_p.h>
+#include <private/qeglfskmsdevice_p.h>
 
 #include <gbm.h>
 
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen.cpp b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen.cpp
index 294b09b178..979c02847e 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen.cpp
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen.cpp
@@ -42,7 +42,7 @@
 #include "qeglfskmsgbmscreen_p.h"
 #include "qeglfskmsgbmdevice_p.h"
 #include "qeglfskmsgbmcursor_p.h"
-#include "qeglfsintegration_p.h"
+#include <private/qeglfsintegration_p.h>
 
 #include <QtCore/QLoggingCategory>
 
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen_p.h b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen_p.h
index ed4b2ec223..2acf5ae2d5 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen_p.h
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/qeglfskmsgbmscreen_p.h
@@ -53,7 +53,7 @@
 // We mean it.
 //
 
-#include "qeglfskmsscreen_p.h"
+#include <private/qeglfskmsscreen_p.h>
 #include <QMutex>
 #include <QWaitCondition>
 
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/eglfs_kms_egldevice.pro b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/eglfs_kms_egldevice.pro
index a6145c07e6..42e4c3d124 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/eglfs_kms_egldevice.pro
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/eglfs_kms_egldevice.pro
@@ -2,8 +2,6 @@ TARGET = qeglfs-kms-egldevice-integration
 
 QT += core-private gui-private eglfsdeviceintegration-private eglfs_kms_support-private kms_support-private edid_support-private
 
-INCLUDEPATH += $$PWD/../../api $$PWD/../eglfs_kms_support
-
 # Avoid X11 header collision, use generic EGL native types
 DEFINES += QT_EGL_NO_X11
 
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldevice.h b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldevice.h
index aa321e0982..77a8a97f97 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldevice.h
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldevice.h
@@ -40,7 +40,7 @@
 #ifndef QEGLFSKMSEGLDEVICE_H
 #define QEGLFSKMSEGLDEVICE_H
 
-#include <qeglfskmsdevice_p.h>
+#include <private/qeglfskmsdevice_p.h>
 
 QT_BEGIN_NAMESPACE
 
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldeviceintegration.h b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldeviceintegration.h
index 7dff46888d..417d8e64ce 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldeviceintegration.h
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldeviceintegration.h
@@ -41,7 +41,7 @@
 #ifndef QEGLFSKMSEGLDEVICEINTEGRATION_H
 #define QEGLFSKMSEGLDEVICEINTEGRATION_H
 
-#include <qeglfskmsintegration_p.h>
+#include <private/qeglfskmsintegration_p.h>
 
 #include <xf86drm.h>
 #include <xf86drmMode.h>
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldevicescreen.h b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldevicescreen.h
index 7cede5afbe..50ff3ad353 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldevicescreen.h
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/qeglfskmsegldevicescreen.h
@@ -40,7 +40,7 @@
 #ifndef QEGLFSKMSEGLDEVICESCREEN_H
 #define QEGLFSKMSEGLDEVICESCREEN_H
 
-#include <qeglfskmsscreen_p.h>
+#include <private/qeglfskmsscreen_p.h>
 
 QT_BEGIN_NAMESPACE
 
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_support/eglfs_kms_support.pro b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_support/eglfs_kms_support.pro
index faad423be1..478f7318f6 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_support/eglfs_kms_support.pro
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_support/eglfs_kms_support.pro
@@ -4,8 +4,6 @@ load(qt_module)
 
 QT += core-private gui-private eglfsdeviceintegration-private kms_support-private edid_support-private
 
-INCLUDEPATH += $$PWD/../../api
-
 # Avoid X11 header collision, use generic EGL native types
 DEFINES += QT_EGL_NO_X11
 
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_support/qeglfskmsscreen.cpp b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_support/qeglfskmsscreen.cpp
index 39c5d317b2..0e7939e137 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_support/qeglfskmsscreen.cpp
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_support/qeglfskmsscreen.cpp
@@ -41,7 +41,7 @@
 
 #include "qeglfskmsscreen_p.h"
 #include "qeglfskmsdevice_p.h"
-#include "qeglfsintegration_p.h"
+#include <private/qeglfsintegration_p.h>
 
 #include <QtCore/QLoggingCategory>
 
diff --git a/src/plugins/platforms/eglfs/eglfs-plugin.pro b/src/plugins/platforms/eglfs/eglfs-plugin.pro
index ec229796e5..90da265000 100644
--- a/src/plugins/platforms/eglfs/eglfs-plugin.pro
+++ b/src/plugins/platforms/eglfs/eglfs-plugin.pro
@@ -11,8 +11,6 @@ SOURCES += $$PWD/qeglfsmain.cpp
 
 OTHER_FILES += $$PWD/eglfs.json
 
-INCLUDEPATH += $$PWD/api
-
 PLUGIN_TYPE = platforms
 PLUGIN_CLASS_NAME = QEglFSIntegrationPlugin
 !equals(TARGET, $$QT_DEFAULT_QPA_PLUGIN): PLUGIN_EXTENDS = -
diff --git a/src/plugins/platforms/eglfs/qeglfsmain.cpp b/src/plugins/platforms/eglfs/qeglfsmain.cpp
index 4f77b7cd17..6a39a21338 100644
--- a/src/plugins/platforms/eglfs/qeglfsmain.cpp
+++ b/src/plugins/platforms/eglfs/qeglfsmain.cpp
@@ -38,7 +38,7 @@
 ****************************************************************************/
 
 #include <qpa/qplatformintegrationplugin.h>
-#include "qeglfsintegration_p.h"
+#include <private/qeglfsintegration_p.h>
 
 QT_BEGIN_NAMESPACE
 
